src/urllib3/util/connection.py

```diff
@@ -58,6 +61,11 @@ def create_connection(
     # The original create_connection function always returns all records.
     family = allowed_gai_family()
 
+    try:
+        host.encode("idna")
+    except UnicodeError:
+        raise LocationParseError("Host '%s' is not a valid IDNA label" % host)
```

Comment:
We can use `return six.raise_from(..., None)` here.

Also since the `UnicodeError` contains a nice description of what occurred can we use that message somehow for `LocationParseError`?

Reply:
I did it as you mentioned.